### PR TITLE
Cognito auth 7/7 - add forgot password + reset password

### PR DESCRIPTION
**Pull request by indiv0**
### Pull Request Description

7th PR for IDE/Cloud authorization with cognito. This PR introduces forgot password / reset password templates + flows + amplify wrappers.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import * as hooks from "../../hooks";
+import * as utils from "../../utils";
+import * as app from "../../components/app";
+import * as icons from "../../components/svg";
+
+// =================
+// === Constants ===
+// =================
+
+const RESET_PASSWORD_QUERY_PARAMS = {
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx:17]: consider merging with `RESET_PASSWORD_QUERY_PARAMS` and `LOGIN_QUERY_PARAMS`. see [this PR comment](https://github.com/enso-org/enso/pull/5864/files#r1133629495)

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx [hunk 2]
```diff
@@ -191,10 +194,17 @@ const setDeepLinkHandler = (
       /** Navigate to a relative URL to handle the confirmation link. */
       const redirectUrl = `${app.CONFIRM_REGISTRATION_PATH}${parsedUrl.search}`;
       navigate(redirectUrl);
-    } else if (isSignOutRedirect(parsedUrl)) {
+    } else if (isSignOutRedirect(parsedUrl) || isLoginRedirect(parsedUrl)) {
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx]: consider a `switch`-`case` on `parsedUrl.pathname` (with extra logic in some `case`s to handle things like `isSignInRedirect`). readability would be about the same, but it should be simpler

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx [hunk 3]
```diff
@@ -191,10 +194,17 @@ const setDeepLinkHandler = (
       /** Navigate to a relative URL to handle the confirmation link. */
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx]: this comment might be redundant

File: app/ide-desktop/lib/content/src/dashboard.css [hunk 4]
```diff
@@ -384,6 +384,9 @@ video {
 .-mt-2 {
   margin-top: -0.5rem;
 }
+.-mt-4 {
```
Comment (wdanilo) [inline app/ide-desktop/lib/content/src/dashboard.css]: I was writing about it in other PR review. Please apply other PR reviews and ping me, I'm skipping reviewing this PR for now.

**Review by wdanilo — CHANGES_REQUESTED** [review]

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx (hunk 1 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx:17]: (rejected for now, there's no real benefit)

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx (hunk 2 above)
Comment (indiv0) [reply app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx]: Fixed.

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx (hunk 3 above)
Comment (indiv0) [reply app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx]: Fixed.

File: app/ide-desktop/lib/content/src/dashboard.css (hunk 4 above)
Comment (indiv0) [reply app/ide-desktop/lib/content/src/dashboard.css]: Fixed.

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts [hunk 5]
```diff
@@ -27,6 +27,11 @@ const MESSAGES = {
         userNotConfirmed: 'User is not confirmed. Please check your email for a confirmation link.',
         incorrectUsernameOrPassword: 'Incorrect username or password.',
     },
+    forgotPassword: {
+        userNotFound: 'User not found. Please register first.',
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts]: User is rather found, as he is sitting in front of the computer ;) This message should probably refer "User name". I know this is a minor thing and it's very picky, but this is user-facing messaging and it's important to be very correct there :)

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts [hunk 6]
```diff
@@ -27,6 +27,11 @@ const MESSAGES = {
...
+        userNotConfirmed:
+            'Cannot reset password for user with unverified email. Please verify your email first.',
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts]: not correct English sentence in user-facing messaging. Missing "an". Even better, you can ask ChatGPT to reformat it nicer :)

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/forgotPassword.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+                <div className="inline-flex items-center justify-center absolute left-0 top-0 h-full w-10 text-gray-400">
+                  <icons.Svg data={icons.PATHS.at} />
+                </div>
+
+                <input
+                  {...bindEmail}
+                  id="email"
+                  type="email"
+                  name="email"
+                  className="text-sm sm:text-base placeholder-gray-500 pl-10 pr-4 rounded-lg border border-gray-400 w-full py-2 focus:outline-none focus:border-blue-400"
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/forgotPassword.tsx]: >100 chars

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                <div className="inline-flex items-center justify-center absolute left-0 top-0 h-full w-10 text-gray-400">
+                  <icons.Svg data={icons.PATHS.at} />
+                </div>
+
+                <input
+                  {...bindEmail}
+                  id="email"
+                  type="email"
+                  name="email"
+                  className="text-sm sm:text-base placeholder-gray-500 pl-10 pr-4 rounded-lg border border-gray-400 w-full py-2 focus:outline-none focus:border-blue-400"
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx]: >100 chars

**Review by wdanilo — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+                <div className="inline-flex items-center justify-center absolute left-0 top-0 h-full w-10 text-gray-400">
+                  <icons.Svg data={icons.PATHS.lock} />
+                </div>
+
+                <input
+                  {...bindCode}
+                  id="code"
+                  type="text"
+                  name="code"
+                  className="text-sm sm:text-base placeholder-gray-500 pl-10 pr-4 rounded-lg border border-gray-400 w-full py-2 focus:outline-none focus:border-blue-400"
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/resetPassword.tsx]: please wrap every long lines in the file

**Review by PabloBuchu — APPROVED** [review]

**Review by somebody1234 — APPROVED** [review]

**Pull request merged**